templates/tftmpl/tfvars.go

```diff
@@ -162,21 +158,13 @@ func nonNullMap(m map[string]string) map[string]string {
 	return m
 }
 
-// baseAddressStr is the raw template following hcat syntax for addresses of
+// serviceBaseTmpl is the raw template following hcat syntax for addresses of
 // Consul services.
-const baseAddressStr = `
+const serviceBaseTmpl = `
 {{- with $srv := service %s }}
-  {{- $last := len $srv | subtract 1}}
-  {{- range $i, $s := $srv}}
+  {{- range $s := $srv}}
   "{{ joinStrings "." .ID .Node .Namespace .NodeDatacenter }}" : {
 {{ HCLService $s | indent 4 }}
-  } {{- if (ne $i $last)}},{{end}}
+  },
```

Comment:
nice! much cleaner to have the dangling comma and remove this bit of template + baseCommaStr  ✨